### ISSUE TITLE
initialization of weight Vector of SparseNetwork

### ISSUE THREAD

**Issue by kordjamshidi** [IllinoisCogComp/saul#351]
I wonder if I do something wrong for the wight update [here](https://github.com/IllinoisCogComp/saul/blob/3b026481f08e8853d01b6a97879b19c6b3209461/saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/classifier/SL_model/Saul_SL_Inference.scala#L108) where I get the index of features from the [lexicon] (https://github.com/IllinoisCogComp/saul/blob/3b026481f08e8853d01b6a97879b19c6b3209461/saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/classifier/SL_model/Saul_SL_Inference.scala#L101) and the weight values are derived from [here](https://github.com/IllinoisCogComp/saul/blob/3b026481f08e8853d01b6a97879b19c6b3209461/saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/classifier/SL_model/Saul_SL_Inference.scala#L87)

**Comment by danyaljj** [comment]
Could you clarify a bit more? Not sure what is the error/exception/weird behavior ...

**Comment by kordjamshidi** [comment]
Yes, there is a weird behavior, I read the (lbjava's) weight vectors of a number of classifiers,  make a global vector (SL's weight vector) of float by appending them .  Then, later I split the global float vector given the original sizes of the classifiers weight vectors and read it back to the weight vectors of the classifiers. Problem: The models evaluation changes. For setting back the weight vectors, I need to have a float and example indexes, as I highlighted the lines above. I have also tried ignoring the feature indexes and just keeping a serial 1 to n indexes which did not give me a consistent evaluation too.  I'll try to give a more clear explanation if this is not clear still.
